embassy-stm32/src/can/enums.rs

```diff
@@ -28,3 +29,19 @@ pub enum BusError {
     ///  At least one of error counter has reached the Error_Warning limit of 96.
     BusWarning,
 }
+impl TryFrom<u8> for BusError {
+    type Error = ();
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
```

Comment:
Returning an error while converting an error seemed a bit convoluted on first sight.

Maybe having `BusError::NoError` and dropping TryFrom is not so bad? But it has it's semantical drawbacks. Unsure how rest of code handles cases like this.

Maybe using From and returning Option<BusError> instead of Result would be simpler and still convey complete meaning?

Reply:
Pushed.
This was only used in one place, where reading the registers anyway so I moved it to peripheral.rs and just made it a vanilla 'method'.